### ISSUE TITLE
Zoom deployment based on mass distribution package

### ISSUE THREAD

**Issue by ygini** [autopkg/homebysix-recipes#99]
For few days now, Zoom provide a mass deployment package conform to Apple's guidelines.

We should move to it 

https://support.zoom.us/hc/en-us/articles/115001799006-Mass-Deployment-with-Preconfigured-Settings-for-Mac

**Comment by gregneagle** [comment]
Looks less nasty, though it's still doing unnecessary crap in the postinstall.

At least it's not installing in the installcheck and it doesn't try to write to the current user's home.

**Comment by gregneagle** [comment]
Some issues still with the "enterprise-friendly" package's post-install:

1) the postinstall script kills the installer process, which causes an error to be reported:
```bash
#kill installer
SAVED_STATE_FILE=~/Library/Saved\ Application\ State/com.apple.installer.savedState/windows.plist
installer_pid=$(ps -x -a -o pid -o command | grep "/System/Library/CoreServices/Installer.app"\
| while read each_p
do
    echo "$each_p"
done \
| grep "MacOS" | cut -d " " -f 1)

echo "installer pid: $installer_pid">>$LOG_PATH
echo "["$(date)"]installer saved state: $SAVED_STATE_FILE">>$LOG_PATH

if [[ -f  $SAVED_STATE_FILE ]] ; then
    rm -rf "$SAVED_STATE_FILE"
    ret=$?
    echo "file exist, delete it: $ret">>$LOG_PATH
fi

if [ $installer_pid > 0 ] ; then
    kill -9 $installer_pid
else
    installer_pid=$(ps -ef | grep "/System/Library/CoreServices/Installer.app"\
    | while read each_p
    do
        echo "$each_p"
    done \
    | grep "MacOS" | cut -d " " -f 2)

    echo "installer pid 2: $installer_pid">>$LOG_PATH
    kill -9 $installer_pid

    installer_pid=$(ps -ef | grep "/System/Library/CoreServices/Installer.app"\
    | while read each_p
    do
        echo "$each_p"
    done \
    | grep "MacOS" | cut -d " " -f 3)

    echo "installer pid 3: $installer_pid">>$LOG_PATH
    kill -9 $installer_pid
fi
```

2) The postinstall _removes_ !!!! any preexisting `/Library/Preferences/us.zoom.config.plist` -- so if you are managing configuration preferences for Zoom by placing/managing that file, the installer kills it.

3) The postinstall _launches_ the Zoom application. If this is supposed to be a "mass-deployment" package, I'm sure the admin wants Zoom to be launched on every single machine in their fleet as they update them.
```bash
#launch zoomus app
echo "find login user: $login_user, $ver">>$LOG_PATH
#if [ $ver != "10.11" ] ; then
#	su $b -c "open $2/Contents/MacOS/ZoomPresence"
#else
#When auto update on 10.11, system always return "The application cannot be opened because its executable is missing." while starting ZR with su command. 
#The su command will success after about 8 seconds after auto update.
for (( i = 0; i < 10; i++ )); do
	start_zr=$(su $login_user -c "open $2")
	ret=$?
	echo "start zr status: $start_zr, $ret">>$LOG_PATH

	if [ $ret -eq 0 ]; then
		break
	fi

	sleep 2
done
```

So I'm not convinced this is an actual improvement over the current Autopkg recipes for Zoom, which already deal with some of the installation issues.

**Comment by ygini** [comment]
@gregneagle thanks for your feedback on this. I will try to work with them to fix this.

**Comment by gregneagle** [comment]
Any update/improvement here? I was trying to figure out why I had started some Zoom_IT recipes but never finished them...

**Comment by homebysix** [comment]
Merged #177.
